### ISSUE TITLE
Non-breaking spaces

### ISSUE THREAD

**Issue by war21x3b** [mwilliamson/python-mammoth#57]
Hi. 
From word documents not migrated non-breaking spaces (nbsp). 
In some cases, this is a big problem. How can we solve this? 
Thanks so much.

**Comment by mwilliamson** [comment]
Could you provide an example document, the output you get (both the
HTML and any messages), and the expected output?

**Comment by war21x3b** [comment]
Example of docx https://yadi.sk/i/mKAPe_YL3U89nT
Example of output html https://yadi.sk/d/QkmfAi323U89xX
And I have not any output.
Thanks so much.

**Comment by mwilliamson** [comment]
I'm not sure I see the issue. Your HTML file, and the result of when I try converting the document myself, has non-breaking spaces in. Note that the non-breaking spaces are unicode characters, not HTML entities.

**Comment by war21x3b** [comment]
Hello Michael.
Thanks for your reply.
I use this code:
`with open(file_for_opening, "rb") as docx_file:
    result = mammoth.convert_to_html(docx_file, style_map=style_map,  ignore_empty_paragraphs=False, convert_image=convert_image)`

**Comment by mwilliamson** [comment]
That code seems fine to me. The output you've provided seems correct to
me, so I'm not sure there's an issue to fix.

**Comment by mwilliamson** [comment]
So far as I can tell, the output you've included has non-breaking
spaces, albeit as Unicode characters rather than HTML entities. Are you
saying that you don't think your output has non-breaking spaces, or
that you'd rather have them as HTML entities?

**Comment by war21x3b** [comment]
Yes, a want to have non-breaking spaces in HTML entities. But I don't see how I can do that. The way I see it now is to replace all non-breaking spaces with some sequence of characters, and replace it with &nbsp in my HTML output. But this is a very bad way.

**Comment by mwilliamson** [comment]
You don't need to replace the non-breaking spaces in your input
document (assuming that's what you mean), you can just replace the
non-breaking spaces in the output with the HTML entity.
